ui/pages/settings/flask/snaps-list-tab/view-snap/view-snap.component.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../../../../components/ui/button';
+import Typography from '../../../../../components/ui/typography';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import {
+  TYPOGRAPHY,
+  COLORS,
+} from '../../../../../helpers/constants/design-system';
+import SnapsAuthorshipPill from '../../../../../components/app/flask/snaps-authorship-pill';
+import Box from '../../../../../components/ui/box';
+import ToggleButton from '../../../../../components/ui/toggle-button';
+import PermissionsConnectPermissionList from '../../../../../components/app/permissions-connect-permission-list/permissions-connect-permission-list';
+import ConnectedSitesList from '../../../../../components/app/connected-sites-list';
+import Tooltip from '../../../../../components/ui/tooltip';
+
+function ViewSnap({
+  snap,
+  onRemove,
+  onToggle,
+  connectedSubjects,
+  onDisconnect,
+}) {
+  const t = useI18nContext();
+
+  return (
+    <div className="snap-settings-page__body">
+      <div className="settings-page__content-row">
+        <div className="settings-page__subheader view-snap-subheader">
+          <Typography
+            className="view-snap-title"
+            variant={TYPOGRAPHY.H3}
+            boxProps={{ textAlign: 'center' }}
+          >
+            {snap.manifest.proposedName}
+          </Typography>
+          <Box className="view-snap-pill-toggle-container">
+            <Box className="view-snap-pill-container" paddingLeft={2}>
+              <SnapsAuthorshipPill
+                packageName={snap.manifest.proposedName}
+                url={snap.manifest.repository.url}
```

Comment:
The `repository` field is technically optional. See the spec: https://github.com/MetaMask/specifications/blob/main/snaps/publishing.md
```suggestion
                url={snap.manifest.repository?.url}
```